xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPWinRenderer.h

```diff
@@ -123,12 +123,11 @@ namespace RETRO
     void RenderInternal(bool clear, uint8_t alpha) override;
 
   private:
-    void CompileOutputShader(ESCALINGMETHOD scalingMethod);
-    void HandleScalingChange();
+    void CompileOutputShaders();
     void Render(CD3DTexture *target);
 
-    std::unique_ptr<CRPWinOutputShader> m_outputShader;
-    ESCALINGMETHOD m_prevScalingMethod = VS_SCALINGMETHOD_AUTO;
+    std::unique_ptr<CRPWinOutputShader> m_outputShaderLinear;
+    std::unique_ptr<CRPWinOutputShader> m_outputShaderNearest;
```

Comment:
How about a map from ESCALINGMETHOD to CRPWinOutputShader? That way we don't have to add a member variable for every scaling method.

Reply:
Yeah I was trying to avoid that and go for the solution with less code, but I'll try it.

Reply:
More code is better than less code if it's scalable and maintainable